service/server/server.go

```diff
@@ -220,11 +223,18 @@ func (s *Server) LogEvent(ctx context.Context, e history.Event) error {
 		return errors.Wrapf(err, "logging event")
 	}
 
-	cfg, err := helper.Config.Get()
-	if err != nil {
-		return errors.Wrapf(err, "getting config")
+	// Override the users's slack settings if an events-url flag is provided.
+	var cfg instance.Config
+	if s.defaultEventsConfig != nil {
+		cfg = *s.defaultEventsConfig
```

Comment:
If you put the instance ID in here you would avoid having to pass it down through umpteen functions.
Though it's a bit more messy to do the event type replacement later.